lib/internal/bootstrap/shadow_realm.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+// In ShadowRealm.
+
+const {
+  prepareShadowRealmExecution,
+} = require('internal/process/pre_execution');
+const {
+  BuiltinModule,
+} = require('internal/bootstrap/realm');
+
+BuiltinModule.setRealmAllowRequireByUsers([
+  'assert',
```

Comment:
What is the criteria of allowing modules in this list? It seems somewhat arbitrary to me if `module` is allowed but `fs` isn't (but you can monkey patch `module` to read files and `process.dlopen()` arbitrary things?)